### PR TITLE
feat: add stream config

### PR DESCRIPTION
**Pull request by Echo-minn**
## Motivation

Read stream api configuration from config file

## Modification

add global context value: streamProviders

### CODE REVIEW

**Comment by fly2tomato** [comment]
the CR of frontend is better reviewed by @jsweber

File: openaoe/frontend/src/components/global-config/global-config.tsx [hunk 1]
```diff
@@ -19,6 +21,15 @@ const GlobalConfig: FC<GlobalInfoProps> = ({ children }) => {
             .then(res => {
                 if (res && res.models) {
                     setModels(res.models);
+                    const streamArray = [];
+                    Object.keys(models).forEach((model) => {
+                        if (models[model].webui.isStream !== false && !streamArray.includes(models[model].provider)) {
+                            streamArray.push(models[model].provider);
+                        }
+                    });
+                    if (Array.isArray(streamArray) && streamArray.length > 0) {
```
Comment (jsweber) [inline openaoe/frontend/src/components/global-config/global-config.tsx:30]: istreamArray不需要判断是否为数组，因为它一定是数组

File: openaoe/frontend/src/components/global-config/global-config.tsx [hunk 2]
```diff
@@ -19,6 +21,15 @@ const GlobalConfig: FC<GlobalInfoProps> = ({ children }) => {
             .then(res => {
                 if (res && res.models) {
```
Comment (jsweber) [inline openaoe/frontend/src/components/global-config/global-config.tsx:22]: res.modules需要判断是否为数组

Comment (Echo-minn) [reply openaoe/frontend/src/components/global-config/global-config.tsx:22]: res.modules是一個json, 我判断一下是否为object吧

**Pull request merged**